tests/charm++/varTRAM/vartest.C

```diff
@@ -0,0 +1,98 @@
+#include <numeric>
+#include <chrono>
+#include <random>
+#include <string>
+#include <tuple>
+#include <map>
+#include <vector>
+struct msg;
+template <typename T>
+class is_PUPbytes;/*
+template <>
+struct is_PUPbytes<msg> {
+  static const bool value = false;
+};*/
+#include "vartest.decl.h"
+CProxy_main mainProxy; //readonly
+struct msg {
+  std::vector<int> data;
+  msg(std::vector<int> s) : data(s) {}
+  friend void operator|(PUP::er& er, msg& m) {
+    er|m.data;
+  }
+  msg() : data() {}
+};
+
+class main : public CBase_main {
+  CProxy_engine engines;
+  int N;
+  int msgcount;
+  int lambda;
+public:
+  main(CkArgMsg* args) {
+    if (args->argc == 4) {
+      mainProxy=thisProxy;
+      N=std::atoi(args->argv[1]);
+      msgcount=std::atoi(args->argv[2]);
+      lambda=std::atoi(args->argv[3]);
+      engines=CProxy_engine::ckNew(N,msgcount,lambda,N);
+      engines.simulate();
+    }
+    else {
+      CkPrintf("Usage: %s N Msgcount lambda [given: %d]\n",args->argv[0], args->argc);
+      CkExit();
+    }
+    delete args;
+  }
+  void startsum(int val) {
+    CkPrintf("Sum of generated data: %d\n",val);
+  }
+  void done(int val) {
+    CkPrintf("Sum of final data: %d\n",val);
```

Comment:
If this is a test and not an example, it should be asserting correctness. We probably also want some timing code for performance checks.